orchestration/dagster_orchestration/hca_manage/snapshot.py

```diff
@@ -50,5 +58,84 @@ def create_snapshot(args: argparse.Namespace, host: str) -> JobId:
 
 
 def remove_snapshot(args: argparse.Namespace, host: str) -> JobId:
-    hca = HcaManage(environment=args.env, data_repo_client=get_api_client(host=host))
+    hca = SnapshotManager(environment=args.env, data_repo_client=get_api_client(host=host))
     return hca.delete_snapshot(snapshot_name=args.snapshot_name, snapshot_id=args.snapshot_id)
+
+
+@dataclass
+class SnapshotManager:
+    environment: str
+    data_repo_client: RepositoryApi
+    dataset: Optional[str] = None
```

Comment:
this seems odd to me. i would expect `dataset` to be mandatory for a child object of datasets like a snapshot.

Reply:
Ah good catch

Reply:
So `dataset` and `data_repo_profile_id` are not required for deletions, that is why they're `Optional`

Reply:
makes sense to me. my worry here going forward is that optional properties instantly double the number of possible combinations of values that a class can have, so it's harder to reason about how a function will behave under all circumstances as more and more optional properties show up, but just the fact that we've split these classes out has drastically reduced that problem already, so it's not nearly as big of a deal.